graylog2-server/src/main/java/org/graylog2/cluster/NodeImpl.java

```diff
@@ -46,9 +46,9 @@ public String getNodeId() {
     }
 
     @Override
-    @JsonProperty("is_master")
-    public boolean isMaster() {
-        return (Boolean) fields.get("is_master");
+    @JsonProperty("is_leader")
+    public boolean isLeader() {
+        return (boolean) fields.get("is_leader");
     }
```

Comment:
I wonder if we should use
`@JsonIgnoreProperties(ignoreUnknown = true)` on NodeImpl, so that old entries with `is_master` don't create a problem?

Reply:
As far as I can see we are not deserializing this with Jackson but simply operate on a `Map<String, Object>` so I don't think the annotation would have any effect.

I'm not sure if we are even serializing nodes anywhere, so probably the Jackson annotations are not needed at all. But just to be sure I added a test to check if we properly serialize nodes. In fact we weren't properly serializing the deprecated `is_master` field anymore, so I fixed that.

Can you please check if this makes sense?

Reply:
yeah, it does :+1: